### ISSUE TITLE
stopImmediatePropogation

### ISSUE THREAD

**Issue by jakearchibald** [tschneidereit/servo-runtime-design#3]
Would this be supported in this model? Seems like it'd be impossible due to the async nature of handlers.

**Comment by tschneidereit** [comment]
You're right, it's not supported in an entirely straight-forward manner: we'd need to keep strict ordering of event delivery, meaning that any subscriber that doesn't consume events in a timely manner would effectively block all other consumers - clearly not desirable.

The way I envision this to work is using a transform stream that only passes on those events that it doesn't want to prevent from propagating. If you're not in a position to set yourself up as a gatekeeper in this way, then I'd argue you also shouldn't be able to prevent other subscribers from receiving the event anyway.
